URI.resolve can throw exception

We are using the library for dispatching webhook events, one of the endpoints returned a redirect response with an invalid location (location with spaces) which throw an exception and crashed the app. We are using version `3.3.15` with `httpclient-backend-zio`, stack trace below:

```
java.lang.IllegalArgumentException: Illegal character in path at index 2: go away !!! (/)
? 	at java.base/java.net.URI.create(URI.java:883)
? 	at java.base/java.net.URI.resolve(URI.java:1066)
? 	at sttp.client3.FollowRedirectsBackend.followRedirect(FollowRedirectsBackend.scala:61)
? 	at sttp.client3.FollowRedirectsBackend.$anonfun$followRedirect$2(FollowRedirectsBackend.scala:48)
? 	at scala.Option.fold(Option.scala:263)
? 	at sttp.client3.FollowRedirectsBackend.followRedirect(FollowRedirectsBackend.scala:44)
? 	at sttp.client3.FollowRedirectsBackend.$anonfun$sendWithCounter$1(FollowRedirectsBackend.scala:29)
? 	at zio.internal.FiberContext.evaluateNow(FiberContext.scala:914)
? 	at zio.internal.FiberContext.$anonfun$evaluateLater$1(FiberContext.scala:776)
? 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
? 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
? 	at java.base/java.lang.Thread.run(Thread.java:834)
? Caused by: java.net.URISyntaxException: Illegal character in path at index 2: go away !!! (/)
? 	at java.base/java.net.URI$Parser.fail(URI.java:2913)
? 	at java.base/java.net.URI$Parser.checkChars(URI.java:3084)
? 	at java.base/java.net.URI$Parser.parseHierarchical(URI.java:3166)
? 	at java.base/java.net.URI$Parser.parse(URI.java:3125)
? 	at java.base/java.net.URI.<init>(URI.java:600)
? 	at java.base/java.net.URI.create(URI.java:881)
```

I think the call to resolve should be inside a try expression or the location has first to be validated.

--- a/async-http-client-backend/zio/src/test/scala/sttp/client3/asynchttpclient/zio/ZioFollowRedirectsBackendTest.scala
+++ b/async-http-client-backend/zio/src/test/scala/sttp/client3/asynchttpclient/zio/ZioFollowRedirectsBackendTest.scala
@@ -1,0 +1,42 @@
+package sttp.client3.asynchttpclient.zio
+
+import org.scalatest.flatspec.AsyncFlatSpec
+import org.scalatest.matchers.should.Matchers
+import sttp.capabilities
+import sttp.client3.impl.zio.{RIOMonadAsyncError, ZioTestBase}
+import sttp.client3._
+import sttp.model.{Header, StatusCode}
+import sttp.monad.MonadError
+import zio.Task
+
+class ZioFollowRedirectsBackendTest extends AsyncFlatSpec with Matchers with ZioTestBase {
+  it should "properly handle invalid redirect URIs" in {
+    val stubBackend: SttpBackend[Task, Any] = new SttpBackend[Task, Any] {
+      override def send[T, R >: capabilities.Effect[Task]](request: Request[T, R]): Task[Response[T]] = {
+        Task.succeed(
+          if (request.uri.toString.contains("redirect"))
+            Response.ok("ok".asInstanceOf[T])
+          else
+            Response.apply(
+              "".asInstanceOf[T],
+              StatusCode.PermanentRedirect,
+              "",
+              List(Header.location("i nvalid redirect"))
+            )
+        )
+      }
+
+      override def close(): Task[Unit] = Task.succeed(())
+      override def responseMonad: MonadError[Task] = new RIOMonadAsyncError[Any]
+    }
+
+    val result: Task[Response[_]] = basicRequest
+      .response(asStringAlways)
+      .get(uri"http://localhost")
+      .send(new FollowRedirectsBackend(stubBackend))
+
+    convertZioTaskToFuture.toFuture(result).map { r =>
+      r.body shouldBe "ok"
+    }
+  }
+}

--- a/core/src/main/scala/sttp/client3/FollowRedirectsBackend.scala
+++ b/core/src/main/scala/sttp/client3/FollowRedirectsBackend.scala
@@ -57,8 +57,7 @@ class FollowRedirectsBackend[F[_], P](
       loc: String
   ): F[Response[T]] = {
     val uri = if (FollowRedirectsBackend.isRelative(loc)) {
-      // using java's URI to resolve a relative URI
-      transformUri(uri"${new URI(request.uri.toString).resolve(loc).toString}")
+      transformUri(request.uri.resolve(uri"$loc"))
     } else {
       transformUri(uri"$loc")
     }

